cluster.go

```diff
@@ -106,7 +106,13 @@ func (c *Cluster) refresh(bg bool) error {
 	var errMsgs []string
 	var oldm, newm [HashSlots][]string
 
-	addrs, _ := c.getNodeAddrs(false)
+	// get all node addrs, including replicas
```

Comment:
Note that this does _not_ return all node addresses, it will return the replicas if it knows about any, otherwise the masters. I'm not sure I like switching the `refresh` to get cluster slots from replicas - it should be ok, but it might need a `READONLY` call to make the replica accept the command (I'm not sure since this is not serving a key, it's an administration command), we'd have to test that and it might differ based on redis version... 

I'll come back to review it when I have a bit more time to get back in context, but I think we should address that in `getNodeAddrs` instead and reuse the `StartupNodes` only if both `c.masters` and `c.replicas` are empty, and  here in `refresh` fall back to using the replicas only if `getNodeAddrs(false)` returns an empty list of addresses.

Reply:
> here in refresh fall back to using the replicas only if getNodeAddrs(false) returns an empty list of addresses.

And making sure the replicas do serve the `CLUSTER SLOTS` command without a `READONLY` request first.

Reply:
Indeed, I made the mistake of `getNodeAddrs(true)` will not return all nodes.

> And making sure the replicas do serve the CLUSTER SLOTS command without a READONLY request first

I did tests on redis 5.0.8 and 6.2.13 clusters. The replica node can correctly return cluster slots information without `READONLY`

So, based on the above, can we try to do this?  At the same time, the semantics of `preferReplicas` can be preserved .
```
	if addrs, _ := c.getNodeAddrs(false); len(addrs) == 0 {
		if addrs, _ = c.getNodeAddrs(true); len(addrs) == 0 {
			addrs = c.StartupNodes
		}
	}
```

Reply:
Yes, I think that would be good.